integration-testing/src/functionalTest/kotlin/kotlinx.atomicfu.gradle.plugin.test/framework/checker/DependenciesChecker.kt

```diff
@@ -24,52 +24,47 @@ private fun GradleBuild.checkAtomicfuDependencyIsAbsent(configurations: List<Str
  * compileClasspath — compile dependencies
  * runtimeClasspath — runtime dependencies
  * apiElements — compile dependencies that will be included in publication
  * runtimeElements — runtime dependencies that will be included in publication
  *
- * The functions below check that `org.jetbrains.kotlinx:atomicfu` dependency is only included in compile configurations.
+ * The functions below check that `org.jetbrains.kotlinx:atomicfu` dependency is not present in the runtime configurations.
  */
 
-// Checks a simple JVM project with a single target
-internal fun GradleBuild.checkJvmCompileOnlyDependencies() {
-    checkAtomicfuDependencyIsPresent(listOf("compileClasspath"), jvmAtomicfuDependency)
+internal fun GradleBuild.jvmCheckNoAtomicfuInRuntimeConfigs() {
     checkAtomicfuDependencyIsAbsent(listOf("runtimeClasspath", "apiElements", "runtimeElements"), jvmAtomicfuDependency)
 }
 
-// Checks JVM target of an MPP project
-internal fun GradleBuild.checkMppJvmCompileOnlyDependencies() {
-    checkAtomicfuDependencyIsPresent(listOf("jvmCompileClasspath"), commonAtomicfuDependency)
-    checkAtomicfuDependencyIsAbsent(listOf("jvmRuntimeClasspath", "jvmApiElements", "jvmRuntimeElements"), commonAtomicfuDependency)
-}
-
-// Checks wasmJs target of an MPP project
-internal fun GradleBuild.checkMppWasmJsImplementationDependencies() {
-    checkAtomicfuDependencyIsPresent(listOf("wasmJsCompileClasspath", "wasmJsRuntimeClasspath"), commonAtomicfuDependency)
+internal fun GradleBuild.mppCheckNoAtomicfuInRuntimeConfigs(targetName: String) {
+    checkAtomicfuDependencyIsAbsent(listOf("${targetName}RuntimeClasspath", "${targetName}ApiElements", "${targetName}RuntimeElements"), commonAtomicfuDependency)
 }
 
-internal fun GradleBuild.checkMppWasmWasiImplementationDependencies() {
-    checkAtomicfuDependencyIsPresent(listOf("wasmWasiCompileClasspath", "wasmWasiRuntimeClasspath"), commonAtomicfuDependency)
-}
-
-// Checks Native target of an MPP project
-internal fun GradleBuild.checkMppNativeCompileOnlyDependencies() {
-    // Here the name of the native target is hardcoded because the tested mpp-sample project declares this target and
-    // KGP generates the same set of dependencies for every declared native target ([mingwX64|linuxX64|macosX64...]CompileKlibraries)
-    checkAtomicfuDependencyIsPresent(listOf("macosX64CompileKlibraries"), commonAtomicfuDependency)
-    checkAtomicfuDependencyIsAbsent(listOf("macosX64MainImplementation"), commonAtomicfuDependency)
+internal fun GradleBuild.mppCheckAtomicfuInApi(targetName: String) {
+    checkAtomicfuDependencyIsPresent(listOf("${targetName}MainApi"), commonAtomicfuDependency)
 }
 
 // Checks Native target of an MPP project
-internal fun GradleBuild.checkMppNativeImplementationDependencies() {
-    checkAtomicfuDependencyIsPresent(listOf("macosX64CompileKlibraries", "macosX64MainImplementation"), commonAtomicfuDependency)
+internal fun GradleBuild.mppNativeCheckAtomicfuInImplementation() {
```

Comment:
Could you please pass the target name for the consistency with other checks?

Also, does it make sense to check other native targets?